✨ Feat : Comment - 생성 기능, 조회 기능 구현

JeongO41 opened the pull request:
## PR 체크사항
PR이 다음 사항을 만족하는지 확인해주세요.

<!-- 
체크하려면 괄호 안에 "x"를 입력하세요. 
각 규칙은 Convention 문서에 있습니다.
PR 제목에 쓰는 prefix는 다음과 같습니다.
🚀 Release
🐛 Fix
✨ Feat
📝 Doc
♻️ Refactor
🔧 Chore
⏪️ Revert
🧪 Test
🎉 Init
-->

- [x] 커밋 제목 규칙
- [x] 커밋 메시지 작성 가이드라인
- [x] 라벨, 담당자, 리뷰어 지정


## PR 타입
어떤 유형의 PR인지 체크해주세요.

<!-- 체크하려면 괄호 안에 "x"를 입력하세요. -->

- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Documentation content changes
- [ ] Other... Please describe:


## PR 설명
Comment 생성 및 조회 기능 구현 하였습니다.


## 작업사항

**[코멘트 생성 기능 관련]**
- CommentCreateRequestDto 작성
- CommentController - PostMapping 로직 작성
- CommentService - createComment 로직 작성
- CommentMapper - toComment 로직 작성


**[코멘트 조회 기능 관련]**
- CommentController - GetMapping 로직 작성
- CommentService - getAllComment 로직 작성
- CommentRepository - 코멘트 변경 순으로 불러오는 메서드 선언부 추가
- CommentMapper - toResponseDto 메서드 추가
- CommentResponseDto 작성


## 변경로직
- Comment 조회 시, 변경순으로 불러오기 위해 TimeStamp 필드 추가하였습니다
- #5 브랜치에서 Member 필드 추가한 내용이 이번 작업에도 반영되었습니다

## 참고사항
- 로그인 시, UserDetailsImpl에서 member 객체를 받는다 가정하여 Controller단에서 member 를 인자로 받는 것으로 작성하였습니다
- 두 가지 기능이 한번에 작성되었는데, 다음부터는 한 가지 기능 단위로 작성하도록 할게요 ^^;;
- 추가로, 현재 gradlew와  gradlew.bat 파일이 함께 올라왔는데 이 부분 PR해도 괜찮은 건지 궁금합니다.!


## PR 수정사항
[생성, 조회 기능 모두 아래 변경 사항 적용]
1. Controller 에서 memberId 가 아닌 member 객체를 전달(to Service)
2. Service 단에서 member.getMemberId() 수행
3. Controller에서 return할때 ResponseDto 가 아닌 ResponseEntity를 반환
4. 위 변경사항(3번) 에 따라 Service 단의 return타입을 ResponseDto 로 변경

close #3

JeongO41 committed:
[feat] Comment - 생성, 조회 기능 구현

[코멘트 생성 기능 관련]
- CommentCreateRequestDto 작성
- CommentController - PostMapping 로직 작성
- CommentService - createComment 로직 작성
- CommentMapper - toComment 로직 작성


[코멘트 조회 기능 관련]
- CommentController - GetMapping 로직 작성
- CommentService - getAllComment 로직 작성
- CommentRepository - 코멘트 변경 순으로 불러오는 메서드 선언부 추가
- CommentMapper - toResponseDto 메서드 추가
- CommentResponseDto 작성

JeongO41 committed:
[feat] Comment - Entity 변경

1. Comment 조회 시, 변경순으로 불러오기 위해 TimeStamp 필드 추가하였습니다
2. #5 브랜치에서 Member 필드 추가한 내용이 이번 작업에도 반영되었습니다

hyunmuam commented on src/main/java/com/team1/TBBCloneCoding/comment/controller/CommentController.java:
```diff
@@ -0,0 +1,44 @@
+    @PostMapping("/project/{projectId}/comment")
+    public ResponseDto createComment(Member member, @PathVariable Long projectId, @RequestBody CommentCreateRequestDto commentCreateRequestDto) {
+
+        Long memberId = member.getMemberId();
```
이부분은 `CommentService`에서 수행시켜도 될 것 같습니다. 비즈니스 로직은 최대한 `CommentService`에서 해주세요.
`Controller`와 `Service`가 각각 어떤 역할을 가지고 있는지 생각하시면 좀 더 편하실 거에요.

JeongO41 replied:
서비스단에서 수행하는 것이 더 적절할 것 같습니다. 수정하였습니다!

hyunmuam commented on src/main/java/com/team1/TBBCloneCoding/comment/controller/CommentController.java:
```diff
@@ -0,0 +1,44 @@
+    @GetMapping("/project/{projectId}/comment")
+    public ResponseDto getAllComment(Member member, @PathVariable Long projectId) {
+
+        Long memberId = member.getMemberId();
```
비즈니스 로직은 최대한 CommentService에서 해주세요.
Controller와 Service가 각각 어떤 역할을 가지고 있는지 생각하시면 좀 더 편하실 거에요.

JeongO41 replied:
마찬가지로 수정하였습니다! 감사해요 재현님!

hyunmuam commented on src/main/java/com/team1/TBBCloneCoding/comment/controller/CommentController.java:
```diff
@@ -0,0 +1,44 @@
+    private final CommentService commentService;
+
+    @PostMapping("/project/{projectId}/comment")
+    public ResponseDto createComment(Member member, @PathVariable Long projectId, @RequestBody CommentCreateRequestDto commentCreateRequestDto) {
```
저희 회의 했을 때 `ResponseDto`가 아닌`ResponseEntity`를 반환하기로 했었죠? 생성과 조회 확인하시고 해당 부분의 선언과 반환 방법 수정 부탁드립니다.

JeongO41 replied:
깜빡해부렀습니다.. 수정했습니다!!

hyunmuam reviewed (commented):
리뷰 확인 부탁드립니다.

JeongO41 committed:
[fix] Comment - 생성, 조회 기능 로직 수정

[생성, 조회 기능 모두 아래 변경 사항 적용]
1. Controller 에서 memberId 가 아닌 member 객체를 전달(to Service)
2. Service 단에서 member.getMemberId() 수행
3. Controller에서 return할때 ResponseDto 가 아닌 ResponseEntity를 반환
4. 위 변경사항(3번) 에 따라 Service 단의 return타입을 ResponseDto 로 변경

seonghyun-1 reviewed (approved):
확인했습니다!

JeongO41 committed:
Merge branch 'develop' into feature-#3
